Framework/DataHandling/src/LoadFITS.cpp

```diff
@@ -185,7 +185,7 @@ void LoadFITS::exec() {
  */
 void LoadFITS::doLoadHeaders(const std::vector<std::string> &paths, std::vector<FITSInfo> &headers, size_t firstIndex,
                              size_t lastIndex) {
-  for (size_t i = firstIndex; i <= lastIndex && i <= headers.size(); ++i) {
+  for (size_t i = firstIndex; i <= lastIndex && i < headers.size(); ++i) {
```

Comment:
There's a subtle potential bug here, if `lastIndex > size()` we will run OOB.
If this is impossible with the current flow, can we add an assert to ensure future developers don't break it. Alternatively, if we don't guarantee this could we explicitly check that `lastIndex < size()`?

Reply:
I see now that `paths.size()` isn't checked against `i`. Anyway, this is a private function that is not used anywhere so I've just deleted it.